SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -627,37 +634,135 @@ void ASpatialDebugger::DrawDebug(UCanvas* Canvas, APlayerController* /* Controll
 
 			if (Actor != nullptr)
 			{
-				FVector ActorLocation = Actor->GetActorLocation();
-
-				if (ActorLocation.IsZero())
+				FVector2D ScreenLocation = ProjectActorToScreen(Actor, PlayerLocation);
+				if (ScreenLocation.IsZero())
 				{
 					continue;
 				}
 
-				if (FVector::Dist(PlayerLocation, ActorLocation) > MaxRange)
-				{
-					continue;
-				}
+				DrawTag(Canvas, ScreenLocation, EntityId, Actor->GetName(), true /*bCentre*/);
+			}
+		}
+	}
+}
 
-				FVector2D ScreenLocation = FVector2D::ZeroVector;
-				if (LocalPlayerController.IsValid())
-				{
-					SCOPE_CYCLE_COUNTER(STAT_Projection);
-					UGameplayStatics::ProjectWorldToScreen(LocalPlayerController.Get(), ActorLocation + WorldSpaceActorTagOffset,
-														   ScreenLocation, false);
-				}
+void ASpatialDebugger::SelectActorToTag(UCanvas* Canvas)
+{
+	if (LocalPlayerController.IsValid())
+	{
+		FVector2D MousePosition;
 
-				if (ScreenLocation.IsZero())
+		if (LocalPlayerController->GetMousePosition(MousePosition.X, MousePosition.Y))
+		{
+			bool bCursorChanged = false;
+			if (CrosshairTexture)
+			{
+				// Display a crosshair icon for the mouse cursor
+				// Offset by half of the texture's dimensions so that the center of the texture aligns with the center of the Canvas.
+				FVector2D CrossHairDrawPosition(MousePosition.X - (CrosshairTexture->GetSurfaceWidth() * 0.5f),
+												MousePosition.Y - (CrosshairTexture->GetSurfaceHeight() * 0.5f));
+
+				// Draw the crosshair at the mouse position.
+				FCanvasTileItem TileItem(CrossHairDrawPosition, CrosshairTexture->Resource, FLinearColor::White);
+				TileItem.BlendMode = SE_BLEND_Translucent;
+				Canvas->DrawItem(TileItem);
+				bCursorChanged = true;
+			}
+
+			if (bCursorChanged)
+			{
+				// Hide the mouse cursor as we will draw our own custom crosshair
+				LocalPlayerController->bShowMouseCursor = false;
+				// Sets back the focus to the game viewport - need to hide mouse cursor instantly
+				FSlateApplication::Get().SetAllUserFocusToGameViewport();
```

Comment:
Why is this necessary? Also, by default slate focus is on the debugger UI while it is open, so I'd expect that we should re-set focus to the UI widget once we exit the actor selection mode.

Reply:
If I don't have that line I get both the mouse cursor and the crosshair until I click on the screen
![image](https://user-images.githubusercontent.com/64085832/99075154-a9a4d580-25c1-11eb-90ba-7ea375a887c5.png)

Reply:
Yea sounds about right. Have you checked to see what happens when you're running e.g. 2 PIE clients? I'd be worried that the "all" in there means it'll affect all clients in some way. If it doesn't though then fine.